src/dive/utils/version_info.cpp

```diff
@@ -194,12 +195,39 @@ std::string GetDeviceLibrariesVersionInfo(const std::string& csv_content)
                            GetSHAString(device_info_map[VersionInfoConstants::kNameSha], kLongSha));
 }
 
+std::filesystem::path ResolvePath(std::string_view file_name)
+{
+    std::vector<std::filesystem::path> search_paths{ DIVE_INSTALL_DIR_PATH };
+    if (auto exe_dir = Dive::GetExecutableDirectory(); exe_dir.ok())
+    {
+        search_paths.push_back(*exe_dir);
+        search_paths.push_back(*exe_dir / "install");
+    }
+    else
+    {
+        std::cerr << exe_dir.status().message() << std::endl;
+    }
+
+    for (const auto& p : search_paths)
+    {
+        const auto potential_path = p / file_name;
+        if (std::filesystem::exists(potential_path))
+        {
+            auto canonical_path = std::filesystem::canonical(potential_path);
+            std::cout << "Found " << file_name << " at " << canonical_path << "\n";
+            return canonical_path;
+        }
+    }
+    std::cerr << "Could not find '" << file_name << "' in any of the search paths.\n";
+    return {};
+}
+
 std::string GetLongVersionString()
```

Comment:
Pass the device libraries and profile plugin path in the parameter, so we don't have repeated logic searching for them at multiple places.
```cpp
struct InstallInfo{
  std::filesystem::path device_libraries_path;
  std::filesystem::path profiling_plugin_path;
};
std::string GetLongVersionString(const InstallInfo&);
absl::StatusOr<std::string> GetDeviceLibraryInfo(const std::filesystem::path& device_libraries_path, std::string_view key);
```